src/test/java/seedu/duke/ui/parsers/ParserTest.java

```diff
@@ -1,107 +1,201 @@
+    @Test
+    public void parse_markCommand_noFlagProvided() {
+        final String testString = "mark 123";
+        try {
+            parser.parseCommand(testString);
+            fail();
+        } catch (ParseException e) {
+            return;
+        } catch (Exception e) {
+            fail();
+        }
+    }
 
-    public ParserTest() {
-        super();
+    @Test
+    public void parse_markCommand_noIndexProvided() {
+        final String testString = "mark /c";
+        try {
+            parser.parseCommand(testString);
+            fail();
+        } catch (ParseException e) {
+            return;
+        } catch (Exception e) {
+            fail();
+        }
     }
 
-    @Override
-    public Command parseCommand(String userInput) throws ModHappyException {
-        return null;
+    @Test
+    public void parse_markCommand_notANumber() {
+        final String testString = "mark /c iamnotanumber";
+        try {
+            parser.parseCommand(testString);
+            fail();
+        } catch (ParseException e) {
+            return;
+        } catch (Exception e) {
+            fail();
+        }
     }
 
     @Test
-    public void sampleTest() throws ModHappyException {
-        // Test parser for "add" parameter
+    public void parse_markCommand_parsedCorrectly() {
+        final String testString = "mark /c 3";
         try {
-            commandFormat = "\\s*(?<flag>(/(m|t)))\\s+(?<argument1>[^\\-]*)"
-                    + "\\s*((?<subFlag>\\-(\\bmod|d\\b))\\s+(?<argument2>.+))*";
-            groupNames.clear();
-
-            // add module with -d
-            groupNames.clear();
-            groupNames.add("flag");
-            groupNames.add("argument1");
-            groupNames.add("subFlag");
-            groupNames.add("argument2");
-            HashMap<String, String> parsedCommand = parseString("/m CS2113T -d Software Engineer");
-            assertEquals("/m", parsedCommand.get("flag"));
-            assertEquals("CS2113T", parsedCommand.get("argument1"));
-            assertEquals("-d", parsedCommand.get("subFlag"));
-            assertEquals("Software Engineer", parsedCommand.get("argument2"));
-
-            // add module without -d
-            groupNames.clear();
-            groupNames.add("flag");
-            groupNames.add("argument1");
-            parsedCommand = parseString("/m CS2113T");
-            assertEquals("/m", parsedCommand.get("flag"));
-            assertEquals("CS2113T", parsedCommand.get("argument1"));
-
-            // add task with -mod
-            groupNames.clear();
-            groupNames.add("flag");
-            groupNames.add("argument1");
-            groupNames.add("subFlag");
-            groupNames.add("argument2");
-            parsedCommand = parseString("/t CS2113T Assignment -mod CS2113T");
-            assertEquals("/t", parsedCommand.get("flag"));
-            assertEquals("CS2113T Assignment", parsedCommand.get("argument1"));
```

Comment:
May I know why is my original test deleted? The ParserTest.java is used to test the Parser class and I put my test of the regex here. Also, I notice that you add some tests for ModHappyParser(), how about putting these in ModHappyParserTest?

Reply:
I removed these because they used the old regex which I rewrote. But now I understand that your original tests were supposed to test Parser functionality alone (that is, to make sure it parses "general" regex correctly), whereas the tests I newly wrote are more specifically geared towards ModHappyParser. We can Parser unit tests again.

Reply:
Will be resolved by #77.

Reply:
Okk, thanks